src/App.tsx

```diff
@@ -1,61 +1,131 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const todoId = todos.map(({ id }) => id);
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [currentTodos, setCurrentTodos] = useState([...todos]);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const addHandler = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title.trim() === '') {
+      setTitleError(true);
+    }
+
+    if (userId === 0) {
+      setUserError(true);
+    }
+
+    if (title.trim() === '' || userId === 0) {
+      return;
+    }
+
+    const newTodo = {
+      id: Math.max(...todoId) + 1,
+      userId,
+      title,
+      completed: false,
+      user: getUser(userId),
+    };
+
+    setCurrentTodos((prevTodos) => {
+      return [...prevTodos, newTodo];
+    });
+
+    setTitle('');
+    setUserId(0);
+    setTitleError(false);
+    setUserError(false);
+  };
+
+  const titleHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
```

Comment:
i think you need to rename from titleHandle to handleTitle for example

Reply:
Agree 👍